CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+
+1. Client (Handled by [clientFileGenerator.ts](./src/generators/clientFileGenerator.ts))
+2. Client Context (Handled by [clientContextFileGenerator.ts](./src/generators/clientContextFileGenerator.ts))
+3. Operation Interfaces (Handled by [operationInterfaceGenerator.ts](./src/generators/operationInterfaceGenerator.ts))
+4. Operations (Handled by [operationGenerator.ts](./src/generators/operationGenerator.ts))
+5. Models (Handled by [modelsGenerator.ts](./src/generators/modelsGenerator.ts))
+6. Parameters (Handled by [parametersGenerator.ts](./src/generators/parametersGenerator.ts))
+7. Index File (Handled by [indexGenerator.ts](./src/generators/indexGenerator.ts))
+8. Metadata (Handled [here](./src/generators/static))
+
+The entry point for this design is at [main.ts](./src/main.ts) which invokes the [typescriptGenerator.ts](./src/typescriptGenerator.ts) and starts the entire generation process. You could find all the options that could be sent to the generator at [autorestOptions.ts](./src/utils/autorestOptions.ts)
+
+On a high level, the entire SDK generation process would be:
+
+Swagger Input -> Autorest Core & Modeler -> CodeModel -> TS/JS SDK Generator -> Generated SDK
+
+Within the TS/JS SDK Generator, it has the following stages:
+
+CodeModel + User Options -> Transform CodeModel -> Generate Client -> Generate Client Context -> Generate Operations/Interfaces -> Generate Models -> Generate Parameters -> Generate Index -> Format & Output Generated SDK
+
+### Steps to clone, build & test
+
+1. Use the following command to clone the Typescript/Javascript SDK generator repository:
+
+```
+git clone https://github.com/Azure/autorest.typescript.git
+```
+
+2. Use the following commands to build the SDK generator:
+
+```
+cd autorest.typescript
+npm install
+npm run build
+```
+
+3. There are 3 test-suites in the generator:
+   1. Unit tests (which could be found at `test/unit/*`)
+   2. Integration tests (which could be found at `test/integration/*`)
+   3. Smoke tests
+4. You can run the Unit tests & Integration tests using the following command:
+
+```
+npm run test
+```
+
+**_Note_**: If your development environment is Windows, then run the command `npm run start-test-server:v2`(in a seperate window) before running `npm run test` and run the command `npm run stop-test-server` after.
+
+5. You can run the Smoke tests using the following command:
+
+```
+npm run clone:specs
+npm run smoke-test
+```
+
+**_Note_**: If the command `npm run clone:specs` errors out, delete the `.tmp` folder and then try again.
+
+### How to add an integration test case
+
+In this section, we would learn the steps to add a integration test case. Why is this important? Whenever you work on adding a feature/fixing a bug, this would probably be your first step. You create a test case and then run it through the generator, see the result, modify the generator, run it again and so on, until you get the desired output.
```

Comment:
NIT:
```suggestion
Whenever you work on adding a feature/fixing a bug, this would probably be your first step. You create a test case and then run it through the generator, see the result, modify the generator, run it again and so on, until you get the desired output.
```